### PR TITLE
Set enable_managed_clusters for older versions of Materialize

### PR DESCRIPTION
**Pull request by antiguru**
### Motivation

Fix upgrade tests broken by #20399.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by antiguru** [comment]
Nightly run in https://buildkite.com/materialize/nightlies/builds/3007.

**Review by philip-stoev — APPROVED** [review]
Thank you!

**Review by philip-stoev — CHANGES_REQUESTED** [review]
Cloudtest is currently failing for an unrelated reason, but you also need this patch there:

```
diff --git a/misc/python/materialize/cloudtest/k8s/environmentd.py b/misc/python/materialize/cloudtest/k8s/environmentd.py
index 03e929e5fe..53716a9cdd 100644
--- a/misc/python/materialize/cloudtest/k8s/environmentd.py
+++ b/misc/python/materialize/cloudtest/k8s/environmentd.py
@@ -211,6 +211,9 @@ class EnvironmentdStatefulSet(K8sStatefulSet):
             ]
         if self._meets_minimum_version("0.63.0-dev"):
             args += ["--secrets-controller=kubernetes"]
+        if self._meets_maximum_version("0.63.99"):
+            args += ["--system-parameter-default=enable_managed_clusters=true"]
+
         container = V1Container(
             name="environmentd",
             image=self.image(
@@ -281,3 +284,18 @@ class EnvironmentdStatefulSet(K8sStatefulSet):
 
         minimum_version = Version.parse(minimum)
         return tag_version >= minimum_version
+
+    def _meets_maximum_version(self, maximum: str) -> bool:
+        """Determine whether environmentd is at most the `maximum` version."""
+
+        # Assume that unstable and development versions, as indicated by a
+        # missing or unparseable tag, are always above the maximum
+        if self.tag is None:
+            return False
+        try:
+            tag_version = Version.parse(self.tag.removeprefix("v"))
+        except ValueError:
+            return False
+
+        maximum_version = Version.parse(maximum)
+        return tag_version >= maximum_version
```

Or something along those lines. Though I could not really test it.

**Review by nrainer-materialize — APPROVED** [review]

File: misc/python/materialize/cloudtest/k8s/environmentd.py [hunk 1]
```diff
@@ -281,3 +284,18 @@ def _meets_minimum_version(self, minimum: str) -> bool:
 
         minimum_version = Version.parse(minimum)
         return tag_version >= minimum_version
+
+    def _meets_maximum_version(self, maximum: str) -> bool:
```
Comment (nrainer-materialize) [inline misc/python/materialize/cloudtest/k8s/environmentd.py:288]: The code is pretty similar to `_meets_minimum_version`. What about extracting a generic function, which takes the fallback value (for `tag is None` and `ValueError`) and the comparison operator?

**Comment by philip-stoev** [comment]
@antiguru would you be able to rebase this and bring it over the line today? If not, I will need to take over as Nightly is very red right now.

**Comment by nrainer-materialize** [comment]
> @antiguru would you be able to rebase this and bring it over the line today? If not, I will need to take over as Nightly is very red right now.

He wrote on Slack if QA can take over because he is out (https://materializeinc.slack.com/archives/C01LKF361MZ/p1691001870581149?thread_ts=1690971142.863509&cid=C01LKF361MZ). Shall I take over?

File: misc/python/materialize/cloudtest/k8s/environmentd.py (hunk 1 above)
Comment (philip-stoev) [reply misc/python/materialize/cloudtest/k8s/environmentd.py:288]: Yes!

File: misc/python/materialize/cloudtest/k8s/environmentd.py [hunk 2]
```diff
@@ -281,3 +284,18 @@ def _meets_minimum_version(self, minimum: str) -> bool:
...
+        # missing or unparseable tag, are always above the maximum
+        if self.tag is None:
+            return False
+        try:
+            tag_version = Version.parse(self.tag.removeprefix("v"))
+        except ValueError:
+            return False
+
+        maximum_version = Version.parse(maximum)
+        return tag_version >= maximum_version
```
[the remark this answers is not in the document]
Comment (philip-stoev) [reply misc/python/materialize/cloudtest/k8s/environmentd.py]: Yes!

File: misc/python/materialize/cloudtest/k8s/environmentd.py (hunk 1 above)
Comment (philip-stoev) [reply misc/python/materialize/cloudtest/k8s/environmentd.py]: Done, refactoried it.

Comment (nrainer-materialize) [reply misc/python/materialize/cloudtest/k8s/environmentd.py]: Thank you!

**Review by philip-stoev — APPROVED** [review]

**Pull request merged**